lib/builders/latexmk.js

```diff
@@ -49,21 +51,22 @@ export default class LatexmkBuilder extends Builder {
   }
 
   constructArgs (filePath, jobname) {
-    const outputFormat = atom.config.get('latex.outputFormat') || 'pdf'
-
     const args = [
       '-interaction=nonstopmode',
       '-f',
       '-cd',
-      `-${outputFormat}`,
       '-file-line-error'
     ]
 
+    const outputFormat = this.getOutputFormatFromMagic(filePath) ||
+      atom.config.get('latex.outputFormat') || 'pdf'
     const enableShellEscape = atom.config.get('latex.enableShellEscape')
     const enableSynctex = atom.config.get('latex.enableSynctex') !== false
-    const engineFromMagic = this.getLatexEngineFromMagic(filePath)
-    const customEngine = atom.config.get('latex.customEngine')
-    const engine = atom.config.get('latex.engine')
+    const engine = this.getLatexEngineFromMagic(filePath) ||
+      atom.config.get('latex.customEngine') ||
+      atom.config.get('latex.engine') || 'pdflatex'
+    const producer = this.getProducerFromMagic(filePath) ||
+      atom.config.get('latex.producer') || 'dvipdf'
```

Comment:
Just as for the output format stuff, extract two methods here; e.g. `getLatexEngine` and `getProducer`.
Also try to avoid the inline truthy conditionals. It's generally much easier to read code that use explicit conditional blocks.